### ISSUE TITLE
New rule suggestion: Place all hooks at the top of the block.

### ISSUE THREAD

**Issue by garyking** [jest-community/eslint-plugin-jest#356]
I would like to suggest a new rule, to place all hook blocks before any other blocks.

So, this is good:

```js
describe('module', () => {
  let a
  
  beforeEach(() => {
    a = new A();
  })
  
  test('function', () => {
    expect(a).not.toBeUndefined();
  })
})
```

And this is bad:

```js
describe('module', () => {
  let a
  
  test('function', () => {
    expect(a).not.toBeUndefined();
  })
  
  beforeEach(() => {
    a = new A();
  })
})
```

**Comment by G-Rath** [comment]
Sounds like it could be useful - I'd think a good target goal would be something like "all hooks must be above the first test case in a block".

PRs welcome 😄
